modules/rosetta/src/test/scala/org/tessellation/rosetta/domain/construction/ConstructionServiceSuite.scala

```diff
@@ -31,44 +32,40 @@ import weaver.scalacheck.Checkers
 
 object ConstructionServiceSuite extends MutableIOSuite with Checkers with TransactionGenerator {
 
-  type Res = (SecurityProvider[IO], KryoSerializer[F])
+  type Res = (SecurityProvider[IO], KryoSerializer[IO])
 
   override def sharedResource: Resource[IO, Res] =
     KryoSerializer
       .forAsync[IO](sharedKryoRegistrar)
       .flatMap(kryo => SecurityProvider.forAsync[IO].map(securityProvider => (securityProvider, kryo)))
 
-  def generateTestTransactions(wantSignedTransaction: Boolean)(implicit S: SecurityProvider[IO], K: KryoSerializer[IO]) = {
-    val getSignedTransaction = (transaction: Hashed[Transaction]) => transaction.signed
-    val getUnsignedTransaction = (transaction: Hashed[Transaction]) => transaction.signed.value
-
-    def txs(transactionConverter: Hashed[Transaction] => AnyRef) =
-      (KeyPairGenerator.makeKeyPair[IO], KeyPairGenerator.makeKeyPair[IO]).tupled.flatMap {
-        case (srcKey, dstKey) =>
-          val srcAddress = srcKey.getPublic.toAddress
-          val dstAddress = dstKey.getPublic.toAddress
-          val txCount = PosInt(100)
-
-          generateTransactions(srcAddress, srcKey, dstAddress, txCount)
-            .flatMap(_.traverse { hashedTransaction =>
-              KryoSerializer[F]
-                .serialize(transactionConverter(hashedTransaction))
-                .map(Hex.fromBytes(_))
-                .map((_, hashedTransaction.hash, hashedTransaction))
-                .liftTo[IO]
-            })
-      }
-
-    if (wantSignedTransaction) {
-      txs(getSignedTransaction)
-    } else {
-      txs(getUnsignedTransaction)
-    }
+  private def getBytes(hashedTransaction: Hashed[DAGTransaction], wantSignedTransaction: Boolean) = if (wantSignedTransaction) {
+    JsonBinarySerializer.serialize(hashedTransaction.signed)
+  } else {
+    JsonBinarySerializer.serialize(hashedTransaction.signed.value)
   }
 
+  def generateTestTransactions(wantSignedTransaction: Boolean)(implicit S: SecurityProvider[IO], K: KryoSerializer[IO]) =
+    (KeyPairGenerator.makeKeyPair[IO], KeyPairGenerator.makeKeyPair[IO]).tupled.flatMap {
+      case (srcKey, dstKey) =>
+        val srcAddress = srcKey.getPublic.toAddress
+        val dstAddress = dstKey.getPublic.toAddress
+        val txCount = PosInt(100)
+
+        for {
+          transactions <- generateTransactions(srcAddress, srcKey, dstAddress, txCount)
+          testValues <- transactions.traverse(hashedTransaction =>
+            getBytes(hashedTransaction, wantSignedTransaction)
+              .asRight[Throwable]
```

Comment:
`ConstructionError` ?
Let's avoid using `Throwable` unless we're forced to adhere to a 3rd party interface.